packages/lib/test/inputGen.test.ts

```diff
@@ -0,0 +1,166 @@
+// @ts-ignore
+import { buildPoseidonReference } from "circomlibjs";
+import {
+  generateMerkleProof,
+  getPublicInputsFromSignature,
+} from "../src/inputGen";
+import {
+  hashEdwardsPublicKey,
+  hexToBigInt,
+  publicKeyFromString,
+} from "../src/utils";
+import { privateKeyToPublicKey } from "../src/ecdsa";
+import { EdwardsPoint } from "../src/babyJubjub";
+
+describe("merkle tree", () => {
+  test("should generate the same merkle root as the circuit", async () => {
+    const pubKeys = [
+      "041941f5abe4f903af965d707182b688bd1fa725fd2cbc648fc435feb42a3794593275a2e9b4ad4bc0d2f3ecc8d23e3cf89da889d7aa35ce33f132d87b5bb5c393",
+      "049ae9f2ec6a4db43f0e081a436f885b0d3f5753a45b00d2f2e3da38956848c4ff0205d89e14a2e36976bfe033407dbce6b48261d84d201277de0c3b82f08ddb09",
+      "041052d6da0c3d7248e39e08912e2daa53c4e54cd9f2d96e3702fa15e77b199a501cd835bbddcc77134dc59dbbde2aa702183a68c90877906a31536eef972fac36",
+      "044d9d03f3266f24777ac488f04ec579e1c4bea984398c9b98d99a9e31bc75ef0f13a19471a7297a6f2bf0126ed93d4c55b6e98ec286203e3d761c61922e3a4cda",
+    ];
+    const index = 2;
+    const poseidon = await buildPoseidonReference();
+
+    const merkleProof = await generateMerkleProof(pubKeys, index);
+
+    const pubKey = pubKeys[index];
+    const pubKeyWeierstrass = publicKeyFromString(pubKey);
+    const pubKeyEdwards = pubKeyWeierstrass.toEdwards();
+    const leaf = await hashEdwardsPublicKey(pubKeyEdwards);
+
+    let node = leaf;
+    for (let i = 0; i < 10; i++) {
+      const otherNode = merkleProof.siblings[i];
+      if (merkleProof.pathIndices[i] % 2 === 0) {
+        node = poseidon([node, otherNode]);
+      } else {
+        node = poseidon([otherNode, node]);
+      }
+      node = hexToBigInt(poseidon.F.toString(node, 16));
+    }
+
+    const expectedRoot = BigInt(
+      "4634016523752195062014614389412713184983003254567244385609775692869553859565"
+    );
+
+    console.log(merkleProof.pathIndices);
+    console.log(merkleProof.siblings);
+    console.log(merkleProof.root);
+    expect(merkleProof.root).toEqual(expectedRoot);
+    expect(node).toEqual(expectedRoot);
+  });
+});
+
+describe("membership proof input generation", () => {
+  test("should generate the correct membership proof inputs 0", () => {
+    const privKey =
+      "0323dbbda9a5aff570d974d71c88334cf99ab9c0455e1d2546ca03ca069eb1e0";
+    const msgHash = BigInt("0");
+    const sig = {
+      r: hexToBigInt(
+        "00EF7145470CEC0B683C629CBA8ED58110000FFE657366F7D5A91F2D149DD8B5"
+      ),
+      s: hexToBigInt(
+        "0370C60A23266F520C56DA088B4C4AFAAAF6BB1993A501980F6D8FB6F343984A"
+      ),
+    };
+
+    const pubKey = privateKeyToPublicKey(hexToBigInt(privKey));
+
+    const expectedPublicInputs = {
```

Comment:
Same comment as other tests, can we have documentation for how these were computed?

Reply:
Will add documents to all the tests